flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -98,12 +102,21 @@ public String toString() {
         .toString();
   }
 
-  /** close all open files and emit files to downstream committer operator */
+  /**
+   * close all open files and emit files to downstream committer operator
+   */
   private void flush() throws IOException {
+    if (writer == null) {
+      return;
+    }
+
     long startNano = System.nanoTime();
     WriteResult result = writer.complete();
     writerMetrics.updateFlushResult(result);
     output.collect(new StreamRecord<>(result));
     writerMetrics.flushDuration(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNano));
+
+    // Set write to null to prevent multiple calls to the complete method of the same write
```

Comment:
nit: maybe comment like this

```
Set writer to null to prevent duplicate flushes in the corner case of prepareSnapshotPreBarrier happening after endInput. 
```

Reply:
Thanks for your review, I have changed.